pkg/api/stream.go

```diff
@@ -17,22 +17,26 @@ package api
 import (
 	"context"
 	"sync"
+	"time"
 )
 
 type SourceTuple interface {
 	Message() map[string]interface{}
 	Meta() map[string]interface{}
+	Timestamp() time.Time
 }
 
 type DefaultSourceTuple struct {
 	Mess map[string]interface{} `json:"message"`
 	M    map[string]interface{} `json:"meta"`
+	Time time.Time              `json:"timestamp"`
 }
 
-func NewDefaultSourceTuple(message map[string]interface{}, meta map[string]interface{}) *DefaultSourceTuple {
+func NewDefaultSourceTuple(message map[string]interface{}, meta map[string]interface{}, timestamp time.Time) *DefaultSourceTuple {
```

Comment:
Let's keep the old function with a default timestamp and create a new function for 3 parameters. Otherwise, users will need to update the customized plugin.